File: Go/1633/B/solution.go
// Author: Shivshankar Pirgonde

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func solve(w *bufio.Writer, s []byte) {
	zeros, ones := 0, 0

	for _, b := range s {
		if b == '0' {
			zeros++
		} else {
			ones++
		}
	}

	ans := zeros

	if ones < zeros {
		ans = ones
	}
	if zeros == ones {
		ans--
	}

	fmt.Fprintln(w, ans)

}

func main() {

	const maxCap = 200005

	sc := bufio.NewScanner(os.Stdin)

	buf := make([]byte, maxCap)
	sc.Buffer(buf, maxCap)

	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	//Read test input using	strconv method
	if !sc.Scan() {
		return
	}

	t, _ := strconv.Atoi(sc.Text())

	for i := 0; i < t; i++ {
		if !sc.Scan() {
			break
		}

		solve(out, sc.Bytes())
	}
}

File: C/1633/B/solution.c
// Author: Shivshankar Pirgonde

#include <stdio.h>

void solve(char s[])
{
    int zeros = 0, ones = 0;

    for (char *p = s; *p != '\0'; p++)
    {
        if (*p == '0')
            zeros++;
        else
            ones++;
    }

    int ans = (zeros < ones) ? zeros : ones;

    if (zeros == ones)
        ans--;

    printf("%d\n", ans);
}

int main()
{
    char s[200005]; 
    int t;

    scanf("%d", &t);

    while (t--)
    {

        scanf("%s", s);
        solve(s);
    }
    return 0;
}